erts/emulator/beam/erl_process.c

```diff
@@ -9051,8 +9119,8 @@ resume_process_1(BIF_ALIST_1)
 {
```

Comment:
I also dislike that the timer gets paused even after all those interested in pausing it, resumed the process. This should be changed. Do we need to add a new option though? I was thinking of adding a new counter in the process structure, similar to `rcount` but with the count of processes that have suspended the proc_timer. When `resume_process()` we know if the caller paused the proc timer as well, so we know if the new counter needs decreasing. When it reaches 0, we could unpause the timer. Would that work?

Reply:
An option is better since:
* You don't need to add anything to the process structure. We try to avoid adding things to the process structure when it is not frequently used since all processes will have to allocate memory for it even when not used. Since you can have a very large amount of processes, this overhead can become significant. There might be exceptions for important things or when it is not possible to do something another way. This is a pure debug feature that can be done some other way, though.
* The unpause could potentially be done too early. The resume operation corresponding to the suspend operation with pause could be done later than other suspend/resume operations which potentially would unpause the timer too early.

I think you should have a counter in the `ErtsMonitorSuspend` structure, though, and throw a `badarg` error exception in the `erlang:resume_process()` BIF if the amount of suspend with pause and suspend without pause do not match with the amount of resume with unpause and resume without unpause.